apps/emqx_rule_engine/src/emqx_rule_runtime.erl

```diff
@@ -166,16 +176,18 @@ select_and_collect([Field|More], Input, {Output, LastKV}) ->
         {nested_put(Key, Val, Output), LastKV}).
 
 %% Filter each item got from FOREACH
--dialyzer({nowarn_function, filter_collection/4}).
-filter_collection(Input, InCase, DoEach, {CollKey, CollVal}) ->
+-dialyzer({nowarn_function, filter_collection/5}).
+filter_collection(RuleId, Input, InCase, DoEach, {CollKey, CollVal}) ->
```

Comment:
If we handle the increasing of the metrics only at the top level `try..catch`, then this additional parameter is not necessary.